ml/BitRateWindow.cc

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "BitRateWindow.h"
+
+using namespace ml;
+
+std::pair<BitRateWindow::Edge, size_t> BitRateWindow::insert(bool Bit) {
```

Comment:
Since you have created the state diagram for this FSM already maybe committing it as `BitRateWindow_FSM.svg` would be best way (most natural place for somebody who needs it to find it).... Extra points for doing it in `plantuml` as that would be easier to update in future and plays nice with git (human readable diffs)

Reply:
I had this in mind. I'll update the diagram because it was outdated (Beware, I only included it in my presentation to explain the concept. The transitions do not reflect 100% what the code really does).

Reply:
Well I will leave it up to you whether you think it is worth it to have and maintain documentation on this and in what form. Just making a suggestion since you already created a diagram for it.

Reply:
Resolving this. I'll add the graph after the MVP.